src/Microsoft.AspNetCore.ResponseCaching/Streams/StreamUtilities.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.ResponseCaching.Internal
+{
+    internal static class StreamUtilities
+    {
+        /// <summary>
+        /// The segment size for buffering the response body in bytes. The default is set to 84 KB.
+        /// </summary>
+        // Internal for testing
+        internal static int BodySegmentSize { get; set; } = 84 * 1024;
```

Comment:
Is there a reason this was set to 86016 (which is just above the LOH limit of 85K)?

The default buffer size for `CopyTo{Async}` [is set to 81920](https://github.com/dotnet/coreclr/blob/b089964f6546a6473e0f5d90d0cae49f22c57455/src/mscorlib/src/System/IO/Stream.cs#L39) which is the largest multiple of 4096 that is smaller than the large object heap threshold. Shouldn't this follow the same limit?

Reply:
Yes it should.

Reply:
https://github.com/aspnet/ResponseCaching/pull/146